pkg/ast/common.go

```diff
@@ -258,19 +258,19 @@ func createMatchFilterCriteria(colName, colValue interface{}, opr LogicalOperato
 	return &criteria
 }
 
-func CreateTermFilterCriteria(colName string, colValue interface{}, opr FilterOperator, qid uint64, cci *CaseConversionInfo) *FilterCriteria {
+func CreateTermFilterCriteria(colName string, colValue interface{}, opr FilterOperator, qid uint64, cci *CaseConversionInfo, valueIsRegex bool) *FilterCriteria {
 	cVal, err := CreateDtypeEnclosure(colValue, qid)
 	if err != nil {
 		log.Errorf("qid=%d, createTermFilterCriteria: error creating DtypeEnclosure for ColValue=%v. Error=%+v", qid, colValue, err)
 	} else {
-		if cci != nil {
+		if cci != nil && !valueIsRegex {
 			cVal.UpdateRegexp(cci.caseInsensitive)
```

Comment:
Move this inside the below if condition. We only need to do this at that place.

Reply:
`ShouldAlsoSearchWithOriginalCase` will ensure that if `valueIsRegex` set to true, it won't be considered as a case-insensitive search.

Reply:
addressed in the latest commit.